src/plugins/ResizeReorder/ReorderHandle.js

```diff
@@ -0,0 +1,241 @@
+   */
+  originalLeft = 0;
+
+  render() {
+    const style = {
+      height: this.props.height,
+    };
+
+    return (
+      <div
+        className={cx({
+          'fixedDataTableCellLayout/columnReorderContainer': true,
+          'fixedDataTableCellLayout/columnReorderContainer/active': false,
+        })}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.props.touchEnabled ? this.onMouseDown : null}
+        onTouchEnd={this.props.touchEnabled ? e => e.stopPropagation() : null}
+        onTouchMove={this.props.touchEnabled ? e => e.stopPropagation() : null}
+        style={style}>
+      </div>
+    );
+  }
+
+  /**
+   *
+   * @param {MouseEvent} event
+   */
+  onMouseDown = (event) => {
+    this.distance = 0;
+    this.animating = true;
+    this.scrollStart = getState().scrollX;
+    this.originalLeft = this.props.left;
+    this.initializeDOMMouseMoveTracker();
+    this.mouseMoveTracker.captureMouseMoves(event);
+    this.updateParentReorderingData({
+      isColumnReordering: true,
+      displacement: 0
+    });
+    this.frameId = requestAnimationFrame(this.updateState);
+  };
+
+  /**
+   * @param {number} deltaX
+   */
+  onMouseMove = (deltaX) => {
+    this.distance += deltaX * (this.props.isRTL ? -1 : 1);
+  };
+
+  onMouseUp = () => {
+    cancelAnimationFrame(this.frameId);
+    this.calculateColumnOrder();
+    this.animating = false;
+    this.frameId = null;
+    this.distance = 0;
+    this.mouseMoveTracker.releaseMouseMoves();
+    this.updateParentReorderingData({
+      isColumnReordering: false,
+      displacement: 0
+    });
+  };
+
+  initializeDOMMouseMoveTracker = () => {
+    this.mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled
+    );
+  };
+
+  updateParentReorderingData = (state) => {
+    this.props.updateParentReorderingData(state);
+  };
+
+  updateState = () => {
+    if (this.animating) {
+      this.frameId = requestAnimationFrame(this.updateState);
+    }
+    this.calculateDisplacementWithScroll();
+  };
+
+  calculateDisplacementWithScroll = () => {
+    const scrollStart = this.scrollStart;
+    let { isFixed, maxScrollX } = this.props;
+    let { scrollX } = getState();
+    let deltaX = this.distance;
+    if (!isFixed) {
+      // Relative dragX position on scroll
+      const dragX = this.originalLeft - scrollStart + deltaX;
+      const availableScrollWidth = this.props.availableScrollWidth;
+      deltaX += scrollX - scrollStart;
+
+      // Scroll the table left or right if we drag near the edges of the table
+      if (dragX > availableScrollWidth - DRAG_SCROLL_BUFFER) {
+        scrollX = Math.min(scrollX + DRAG_SCROLL_SPEED, maxScrollX);
+      } else if (dragX <= DRAG_SCROLL_BUFFER) {
+        scrollX = Math.max(scrollX - DRAG_SCROLL_SPEED, 0);
+      }
+      this.props.scrollToX(scrollX);
+    }
+    this.updateParentReorderingData({ displacement: deltaX });
```

Comment:
@pradeepnschrodinger Should we enclose it in a closure to compare with the previous distance and only update parent when it has been changed ?

Reply:
you can have the check on the parent

Reply:
Done